examples/python/ex3-volume.py

```diff
@@ -0,0 +1,426 @@
+        "-p",
+        "--solution-degree",
+        type=int,
+        default=4,
+        help="Solution polynomial degree (default: 4)",
+        dest="sol_degree",
+    )
+    parser.add_argument(
+        "-q",
+        "--quadrature-points",
+        type=int,
+        default=6,
+        help="Number of quadrature points in 1D (default: 6)",
+        dest="num_qpts",
+    )
+    parser.add_argument(
+        "-s",
+        "--problem-size",
+        type=int,
+        default=1000,
+        help="Approximate problem size (default: 1000)",
+        dest="problem_size",
+    )
+    return parser.parse_args()
+
+
+def run_example_3(args):
+    """Run Example 3.
+
+    Args:
+        args: Parsed command line arguments
+    """
+    # Process command line arguments
+    dim = args.dim
+    mesh_degree = args.mesh_degree
+    sol_degree = args.sol_degree
+    num_qpts = args.num_qpts
+    prob_size = args.problem_size
+    ceed_spec = args.ceed_spec
+
+    # Set up the mesh
+    num_xyz = get_cartesian_mesh_size(dim, sol_degree, prob_size)
+    mesh_degree = max(1, mesh_degree)  # Mesh degree must be at least 1
+
+    # Print summary
+    print(f"Selected options: mesh degree = {mesh_degree}, solution degree = {sol_degree}, "
+          f"number of 1D quadrature points = {num_qpts}")
+    print(f"Mesh size: {' x '.join(map(str, num_xyz))} elements")
+
+    # Set up libCEED
+    ceed = libceed.Ceed(ceed_spec)
+
+    # Build CeedElemRestriction objects describing the mesh and solution fields
+    mesh_restriction, _, mesh_size = build_cartesian_restriction(
+        ceed, dim, num_xyz, mesh_degree, dim, num_qpts)
+    sol_restriction, q_data_restriction, sol_size = build_cartesian_restriction(
+        ceed, dim, num_xyz, sol_degree, 1, num_qpts)
+
+    # Create Vectors
+    mesh_coords = ceed.Vector(mesh_size)
+    q_data_comp = 7 if dim == 3 else (4 if dim == 2 else 2)
+    q_data = ceed.Vector(q_data_comp * num_qpts**dim * np.prod(num_xyz))
+    x = ceed.Vector(sol_size)
+    ones = ceed.Vector(sol_size)
+    vol = ceed.Vector(1)
+
+    # Set up mesh coordinates
+    set_cartesian_mesh_coords(dim, num_xyz, mesh_degree, mesh_coords)
+    exact_volume = transform_mesh_coords(dim, mesh_size, mesh_coords)
+
+    # Create bases
+    mesh_basis = ceed.BasisTensorH1Lagrange(dim, dim, mesh_degree + 1, num_qpts,
+                                            GAUSS)
+    sol_basis = ceed.BasisTensorH1Lagrange(dim, 1, sol_degree + 1, num_qpts,
+                                           GAUSS)
+
+    # Create QFunction context
+    build_ctx = ceed.QFunctionContext()
+    ctx_data = np.array([dim, dim], dtype=np.int32)
+    build_ctx.set_data(ctx_data)
+
+    # Load QFunctions
+    qfs = load_qfs_so()
+    file_dir = os.path.dirname(os.path.abspath(__file__))
+
+    # Create the QFunctions
+    qf_setup = ceed.QFunction(1, qfs.wrap_build_mass_diff,
+                              os.path.join(file_dir, "ex3-volume.h:build_mass_diff"))
```

Comment:
```suggestion
    qf_setup = ceed.QFunction(1, qfs.build_mass_diff,
                              os.path.join(file_dir, "ex3-volume.h:build_mass_diff"))
```
I'm not sure where the `wrap_` part is coming from but I'd rather it wasn't there?

Reply:
> I'm not sure where the `wrap_` part is coming from but I'd rather it wasn't there?

The end result is that users of the Python code will never see the wrap_ prefix, while we maintain the necessary technical implementation details in the C layer. This is due to Python bindings for C code.

Reply:
But this is not present in our test suite which uses the same process